middlewares/src/main/java/com/jinyframework/middlewares/jwt/Jwt.java

```diff
@@ -0,0 +1,175 @@
+    public static AuthComponent newAuthComponent(@NonNull Config config) {
+        if (config.secretKey == null) {
+            throw new Exception("Secret key not provided");
+        }
+        val builder = config.toBuilder();
+
+        val secKeyObj = Keys.hmacShaKeyFor(config.secretKey.getBytes(StandardCharsets.UTF_8));
+        builder.secKeyObj(secKeyObj);
+
+        if (config.userKey == null) {
+            builder.userKey(Config.userKeyDefault);
+        }
+        if (config.userRetriever == null) {
+            final UserRetriever retriever = (ctx, claims) -> String.valueOf(claims.get(Claims.SUBJECT));
+            builder.userRetriever(retriever);
+        }
+
+        if (config.authenticator == null) {
+            final Authenticator authenticator = ctx -> new HashMap<>();
+            builder.authenticator(authenticator);
+        }
+        val finalConfig = builder.build();
+
+        return AuthComponent.builder()
+                .handleVerify(verifyHandler(finalConfig))
+                .handleLogin(loginHandler(finalConfig))
+                .build();
+    }
+
+    private static String extractToken(AbstractRequestBinder.Context ctx) {
+        val header = ctx.headerParam(authHeaderKey);
+        return header.startsWith(authHeaderPrefix) ? header.substring(authHeaderPrefix.length()) : null;
+    }
+
+    private static Map<String, Object> verifyToken(SecretKey key, String token) {
+        return Jwts.parserBuilder().setSigningKey(key)
+                .build()
+                .parseClaimsJws(token).getBody();
+    }
+
+    private static Handler verifyHandler(@NonNull Config config) {
+        return ctx -> {
+            val tokStr = extractToken(ctx);
+            if (tokStr == null) {
+                return HttpResponse.of("Authentication failed", HttpURLConnection.HTTP_UNAUTHORIZED);
+            }
+            final Map<String, Object> claims;
+            try {
+                claims = verifyToken(config.secKeyObj, tokStr);
+            } catch (Exception e) {
+                return HttpResponse.of(e.toString(), HttpURLConnection.HTTP_UNAUTHORIZED);
+            }
+            val user = config.userRetriever.retrieveUser(ctx, claims);
+            ctx.setDataParam(config.userKey, user);
+            return HttpResponse.next();
+        };
+    }
+
+    private static Handler loginHandler(@NonNull Config config) {
+        return ctx -> {
+            val claims = config.authenticator.authenticate(ctx);
+            if (claims == null) {
+                return HttpResponse.of("Authentication failed", HttpURLConnection.HTTP_UNAUTHORIZED);
+            }
+            val token = Jwts.builder()
+                    .addClaims(claims)
+                    .signWith(config.secKeyObj)
+                    .compact();
+            val authHeaderVal = authHeaderPrefix + token;
+            ctx.getResponseHeaders().put(authHeaderKey, authHeaderVal);
+            return HttpResponse.of("Authentication successful");
```

Comment:
User might need to custom this part (add custom cookie / return token in HTTP body)